_search-plugins/concurrent-segment-search.md

```diff
@@ -30,8 +32,37 @@ By default, concurrent segment search is disabled on the cluster. You can enable
 The index-level setting takes priority over the cluster-level setting. Thus, if the cluster setting is enabled but the index setting is disabled, then concurrent segment search will be disabled for that index. Because of this, the index-level setting is not evaluated unless it is explicitly set, regardless of the default value configured for the setting. You can retrieve the current value of the index-level setting by calling the [Index Settings API]({{site.url}}{{site.baseurl}}/api-reference/index-apis/get-settings/) and omitting the `?include_defaults` query parameter.
 {: .note}
 
-To enable concurrent segment search for all indexes in the cluster, set the following dynamic cluster setting:
+Both the cluster and index level `search.concurrent_segment_search.mode` settings accept the following values:
+
+`all`: Enables concurrent segment search across all search requests. This is equivalent to setting `search.concurrent_segment_search.enabled: true`.
+
+`none`: Disables concurrent segment search for all search requests, effectively turning off the feature. This is equivalent to `search.concurrent_segment_search.enabled: false` setting. This is the **default** behavior.
+
+`auto`: In this mode, OpenSearch will use the pluggable `ConcurrentSearchRequestDecider` to decide whether to use concurrent or sequential path for the search request based on the query evaluation and whether aggregations are present in the request. By default, if there are no deciders configured by any plugin, then the decision to use concurrent search will be made based on the presence of aggregations in the request.
+For more information about the pluggable decider semantics see [Pluggable concurrent search deciders](#pluggable-concurrent-search-deciders-concurrentsearchrequestdecider)
+
+To enable concurrent segment search for all indexes in the cluster with `auto` mode, set the following dynamic cluster setting:
```

Comment:
nit: I think lets give example about `all` mode instead of `auto` since its simpler to understand and is equivalent to previous example of `true`